### PR TITLE
expression: refactor scalarFuncToPBExpr

### PR DESCRIPTION
**Pull request by zz-jason**
refactor function "scalarFuncToPBExpr":
- use `canFuncBePushed` to determine whether a function can be pushed
- we can further remove `tipb.ExprType_Or` and so on

### CODE REVIEW

**Comment by zz-jason** [comment]
@winoros @XuHuaiyu PTAL

**Comment by winoros** [comment]
Still failed unit-test

File: expression/expr_to_pb.go [hunk 1]
```diff
@@ -198,124 +198,34 @@ func (pc PbConverter) columnToPBExpr(column *Column) *tipb.Expr {
...
-		tp = tipb.ExprType_NullEQ
-	}
-	return pc.convertToPBExpr(expr, tp)
-}
 
-func (pc PbConverter) likeToPBExpr(expr *ScalarFunction) *tipb.Expr {
-	if !pc.client.IsRequestTypeSupported(kv.ReqTypeSelect, int64(tipb.ExprType_Like)) {
+	// check whether this function has ProtoBuf signature.
+	pbCode := expr.Function.PbCode()
+	if pbCode <= 0 {
```
Comment (zimulala) [inline expression/expr_to_pb.go]: Why `pbCode` can not be equal to 0?

File: expression/expr_to_pb.go [hunk 2]
```diff
@@ -198,124 +198,34 @@ func (pc PbConverter) columnToPBExpr(column *Column) *tipb.Expr {
...
-	case ast.GT:
-		tp = tipb.ExprType_GT
-	case ast.NullEQ:
-		tp = tipb.ExprType_NullEQ
-	}
-	return pc.convertToPBExpr(expr, tp)
-}
 
-func (pc PbConverter) likeToPBExpr(expr *ScalarFunction) *tipb.Expr {
-	if !pc.client.IsRequestTypeSupported(kv.ReqTypeSelect, int64(tipb.ExprType_Like)) {
```
Comment (zimulala) [inline expression/expr_to_pb.go]: Why remove this check?

File: expression/expr_to_pb.go (hunk 1 above)
Comment (zz-jason) [reply expression/expr_to_pb.go]: means it is not set.

File: expression/expr_to_pb.go (hunk 2 above)
Comment (zz-jason) [reply expression/expr_to_pb.go]: in the old code logic, it was checked in `convertToPBExpr`, in the new code logic, it is checked in `canFuncBePushed`

**Pull request merged**